docs-user/guide-perf-profiling.md

```diff
@@ -0,0 +1,43 @@
+# Profiling with Linux perf
+
+You can use perf.html to view Linux 'perf' profiles in a much more interactive manner than "perf report" and the like.  Perf has far lower overhead when capturing than the native Gecko Profiler does, especially when we want to look at all threads.
+
+(Importing other profile sources is being investigated, such as in Issue #1138, PR #1065 and PR #1260.)
+
+# Capture
+(This assumes you've installed perf, and set /proc/sys/kernel/perf_event_paranoid to 1 or less.)
+
+Capture a profile (of any program) with:
+```bash
+perf record -g -F 999 program options
+```
+For Firefox, this would be:
+```bash
+perf record -g -F 999 firefox -P profile -no-remote
+```
+
+Alternatively, capture a specific process with
+```bash
+perf record -g -F 999 -p <PID>
+```
+
+After exiting perf, process the perf data into something perf.html can read with:
+```bash
+perf script -F +pid >/tmp/test.perf
+```
+
+You can record perf traces as root, and potentially get more data on the code's execution.  However, firefox will block running as root on Linux; to work around this use:
```

Comment:
Can you add the additional information from the newsgroup thread? More specifically, that "more data" means kernel callstacks, and that setting `kernel.perf_event_paranoid` to 1 or lower is an alternative way of getting this information without using sudo.